### PR TITLE
Singularity Container

### PR DESCRIPTION
**Pull request by ax3l**
Adds a [Singularity container](http://singularity.lbl.gov/). Testing for automated deployment to [Singularity hub](https://www.singularity-hub.org).

### CODE REVIEW

**Comment by ax3l** [comment]
@C0nsultant feel free to merge this PR already, I just need a first draft for Singularity Hub. Ignore the inline comments.

File: Singularity [hunk 1]
```diff
@@ -0,0 +1,44 @@
+Bootstrap: docker
+From: ubuntu:devel
+# Requires: Singularity 2.3+
```
Comment (ax3l) [inline Singularity:3]: Don't find a way to express this programmatically... Asked on slack.

File: Singularity [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+%post
+apt-get update && \
+apt-get install -y --no-install-recommends \
+    cmake \
+    ipython3 \
+    pybind11-dev \
+    libhdf5-openmpi-dev \
+    libopenmpi-dev && \
+rm -rf /var/lib/apt/lists/*
+# libadios-openmpi-dev
```
Comment (ax3l) [inline Singularity]: `libadios` is currently only provided as non-mpi variant in Debian/Ubuntu. Opened upstream request in https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=900804

File: Singularity [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    -DopenPMD_USE_MPI=ON \
+    -DopenPMD_USE_HDF5=ON \
+    -DopenPMD_USE_ADIOS1=OFF \
+    -DopenPMD_USE_ADIOS2=OFF \
+    -DopenPMD_USE_PYTHON=ON \
+    -DPYTHON_EXECUTABLE=$(which python)
+make
+make install
+
+#%test
```
Comment (ax3l) [inline Singularity]: for later: probably have to change dirs again and need to keep the build dir for that... maybe we don't want that.

**Review by anokfireball — APPROVED** [review]

File: Singularity (hunk 1 above)
Comment (ax3l) [reply Singularity:3]: -> check `${SINGULARITY_version}` in `%post` phase

**Pull request merged**